var/spack/repos/builtin/packages/mofem-cephas/package.py

```diff
@@ -0,0 +1,106 @@
+##############################################################################
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+
+class MofemCephas(CMakePackage):
+    """mofem-cephas core library"""
+
+    extendable = True
+
+    @property
+    def root_cmakelists_dir(self):
+        """The relative path to the directory containing CMakeLists.txt
+
+        This path is relative to the root of the extracted tarball,
+        not to the ``build_directory``. Defaults to the current directory.
+
+        :return: directory containing CMakeLists.txt
+        """
+        return os.path.join(self.stage.source_path, 'mofem')
+
+    homepage = "http://mofem.eng.gla.ac.uk"
+    url = "https://likask@bitbucket.org/likask/mofem-cephas.git"
+
+    version('0.8.1', git='https://likask@bitbucket.org/likask/mofem-cephas.git', tag='v0.8.1', submodules=True)
+    version('0.8.0', git='https://likask@bitbucket.org/likask/mofem-cephas.git', tag='v0.8.0', submodules=True)
+    version('0.7.29', git='https://likask@bitbucket.org/likask/mofem-cephas.git', tag='v0.7.29')
+    version('0.7.28', git='https://likask@bitbucket.org/likask/mofem-cephas.git', tag='v0.7.28')
+    version('0.7.27', git='https://likask@bitbucket.org/likask/mofem-cephas.git', tag='v0.7.27')
+    version('develop', git='https://likask@bitbucket.org/likask/mofem-cephas.git', branch='develop')
+
+    # Variants of packages installed as extensions with MoFEM, to indicate this 
+    # specific type of variant prefox with_ is added.
+    variant('with_adol-c', default=True,
```

Comment:
please remove `with_` part, just call it `adol-c`, same below.

Reply:
I would like to keep 'with_' if there is some very strict policy which forbid me doing that. Note that this variant indicates that code, `adol-c` is installed internally not by spack. For example, if I have variant `slepc` installed with Spack I do not add the prefix 'with'. 

That convention enables me to have `with_med`  and  `med`, first installed from source internally as an extension in MoFEM, second is installed with Spack and linked to MoFEM.

Reply:
oh, but generally we do NOT install anything internally. That's somewhat against the whole purpose of Spack. If you look at PETSc, none of its capabilities to compile dependencies is used. 

I would strongly advice against building anything internally.

Reply:
to give an example, what if somebody has a package that uses `mfem` with `adol-c` **and** links against `deal.II+adol-c`. Now you have a problem of having two adol-c's.

Reply:
Fully agree. It makes sense. 

That is the only temporary solution. I am in process of transforming my cmake files to for spack. 

In near future, I going to drop other methods of installation and stick only to Spack. However, for a start, I need something up and running.  It is tricky because for example, I have modified code for tetgen and I not running official code.

I need adol-c 2.5.6, versions above 2.6: working by I found out that for long runs there are some problems on some systems. I do not know why. In Spack currently is version 2.6:

Could you allow for now for this temporary solution?  I know that is not what it should be, I don't like it as well.

Reply:
> I have modified code for tetgen and I not running official code.

you can also apply patches to dependencies from your package, look at `boost` in `dealii`. 

> I need adol-c 2.5.6

maybe it's easier to just add `2.5.6` to spack?